Svc/TraceFileLogger/TraceFileLogger.cpp

```diff
@@ -0,0 +1,238 @@
+// ======================================================================
+// \title  TraceFileLogger.cpp
+// \author sreddy
+// \brief  cpp file for TraceFileLogger component implementation class
+// ======================================================================
+
+#include "Svc/TraceFileLogger/TraceFileLogger.hpp"
+#include "FpConfig.hpp"
+#include <Fw/Types/Assert.hpp>
+#include <Os/File.hpp>
+#include <Os/FileSystem.hpp>
+#include <limits>
+#include <cstring>
+#include <cstdio>
+#include <Fw/Types/StringUtils.hpp>
+#include <Fw/Types/BasicTypes.hpp>
+#include <Fw/Trace/TraceEntry.hpp>
+
+namespace Svc {
+
+    // ----------------------------------------------------------------------
+    // Component construction and destruction
+    // ----------------------------------------------------------------------
+
+    TraceFileLogger::TraceFileLogger(const char* const compName) : 
+    TraceFileLoggerComponentBase(compName),
+    m_log_file(),
+    m_mode(CLOSED),
+    m_fileName(),
+    m_maxFileSize(0),
+    m_byteCount(0),
+    m_log_init(false),
+    m_traceFilter(0)
+    {
+       //Initialize data file
+       this->m_enable_trace = (FW_TRACE_RECORD_TRACE == true) ? true : false;
+       ::memset(m_file_data, 0, sizeof(m_file_data));
+       
+       //Set data and size for file buffer
+       this->m_file_buffer.setData(m_file_data);
+       this->m_file_buffer.setSize(FW_TRACE_MAX_SER_SIZE);
+    
+       //Set array to process trace id filtering
+       this->filterTraceId.set_array(traceId_array,FILTER_TRACE_ID_SIZE);
+    }
+
+    TraceFileLogger::~TraceFileLogger() {
+        if(OPEN == this->m_mode) {
+            //Close file
+            this->m_log_file.close();
+            this->m_mode = CLOSED;
+        }
+    }
+
+    void TraceFileLogger :: 
+        init(FwSizeType queueDepth, FwEnumStoreType instance)
+    {
+            TraceFileLoggerComponentBase::init(queueDepth,instance);
+    }
+
+    void TraceFileLogger::set_log_file(const char* fileName, const U32 maxSize) {
+        
+        FW_ASSERT(fileName != nullptr);
+
+        if(this->m_enable_trace == true) {
```

Comment:
On initialization, make sure the filename and size are stored and file is opened. When enable/disable command is sent, make sure the file writes continuously, on disable flush the file.